phac-nml/irida-next
# Multi level menu component

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in a multi level menu component to the sidebar component to display multi level menus. **NOTE: Currently just working on click of the multi level menu button/label**

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/25466211/edddb8f9-cb1d-4645-bc5b-7a468bb5bf5d)

![image](https://github.com/phac-nml/irida-next/assets/25466211/53abf77b-a386-458f-a981-14fbfb6ed154)


## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Open up `layouts/prpjects.html.erb
2. Replace the `render section.with_item` for the `Settings` link with the below

```
<%= render section.with_multi_level_menu(title: "Settings") do |mm| %>
            <% if allowed_to?(:update?, @project) %>
              <% mm.with_menu_item(
                url: project_edit_path(@project),
                icon: "cog_6_tooth",
                label: "General",
                selected: @current_page == t(:"projects.sidebar.settings").downcase
              ) %>
            <% end %>
            <% mm.with_menu_item(
              url: project_edit_path(@project),
              icon: "cog_6_tooth",
              label: "Bot Accounts",
              selected: @current_page == t(:"projects.sidebar.settings").downcase
            ) %>
          <% end %>
        <% end %>
```
3. You should see the `Settings` sidebar menu item has a downward facing caret
4. Click the `Settings` item and you should see a menu expand below it to show the `General` and `Bot Accounts` links

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
